components/validator/l2_output_submitter.go

```diff
@@ -168,11 +179,21 @@ func (l *L2OutputSubmitter) trySubmitL2Output() error {
 }
 
 // canSubmit checks if submission interval has elapsed and selected for next validator.
-func (l *L2OutputSubmitter) canSubmit(ctx context.Context) (*big.Int, bool, error) {
-	currentBlockNumber, nextBlockNumber, err := l.fetchBlockNumbers(ctx)
+func (l *L2OutputSubmitter) canSubmit() (*big.Int, bool, error) {
+	hasEnoughDeposit, err := l.checkDeposit()
 	if err != nil {
 		return nil, false, err
 	}
+	if !hasEnoughDeposit {
+		l.retryAfter(1)
```

Comment:
What if we use `time.Duration` as the argument to indicate whether it's in seconds or not? It would make it easier to understand when reading the code or following the function calls.

```go
func (l *L2OutputSubmitter) retryAfter(d time.Duration) {
	time.AfterFunc(d, func() {
		l.submitChan <- struct{}{}
	})
}
```